packages/hardhat/contracts/CoinEngine.sol

```diff
@@ -59,6 +59,8 @@ contract MyUSDEngine is Ownable {
 
     function setStaking(address stakingAddress) external onlyOwner {
         i_staking = Staking(stakingAddress);
+        // Set the engine address in the staking contract
+        i_staking.setEngine(address(this));
     }
 
     function setBorrowRate(uint256 newRate) external onlyOwner {
```

Comment:
To enforce that the savings rate is never greater than the borrow rate we should add a check here and revert if it's higher. (This is only because of my suggested edit in _accrueInterest)